File: try-catch.py
#!/usr/bin/env python3

# Cross a Try Statement with a Termination Handler

from time import thread_time_ns


class NotRaisedException(Exception):
    pass


def main(argv):
    times = 1;
    if 1 < len(argv):
        times = int(argv[1])

    start_time = thread_time_ns()
    for count in range(times):
        try:
            pass
        except NotRaisedException:
            pass

    end_time = thread_time_ns()
    print('Run-Time (s): {:.1f}'.format((end_time - start_time) / 1_000_000_000.))


if '__main__' == __name__:
    import sys
    main(sys.argv)

File: try-catch.cpp
// Cross a Try Statement with a Termination Handler
#include <chrono>
#include <cstdlib>
#include <exception>
#include <iostream>
#include <iomanip>

using namespace std;
using namespace std::chrono;

struct NotRaisedException : public std::exception {};

int main(int argc, char * argv[]) {
	unsigned int times = 1;
	volatile bool should_throw = false;
	if (1 < argc) {
		times = strtol(argv[1], nullptr, 10);
	}

	time_point<steady_clock> start_time = steady_clock::now();
	for (unsigned int count = 0 ; count < times ; ++count) {
		try {
			asm volatile ("# try block");
			if (should_throw) {
				throw NotRaisedException();
			}
		} catch (NotRaisedException &) {
			asm volatile ("# catch block");
		}
	}
	time_point<steady_clock> end_time = steady_clock::now();
	nanoseconds duration = duration_cast<nanoseconds>(end_time - start_time);
	cout << "Run-Time (s): " << fixed << setprecision(1) << duration.count() / 1'000'000'000. << endl;
}
